File: threader/src/main/java/com/walhalla/boilerplate/domain/repository/CrudRepository.java
package com.walhalla.boilerplate.domain.repository;

import com.walhalla.boilerplate.domain.repository.base.Repository;

import java.util.List;

/**
 * A sample repository with CRUD operations on a model.
 */
public interface CrudRepository<T> extends Repository {

    boolean insert(T model);

    boolean update(T model);

    T get(long id);

    boolean delete(T model);

    List<T> query();
}

File: threader/src/main/java/com/walhalla/boilerplate/domain/repository/CrudRepository.kt
package com.walhalla.boilerplate.domain.repository

import com.walhalla.boilerplate.domain.repository.base.Repository

/**
 * A sample repository with CRUD operations on a model.
 */
interface CrudRepository<T> : Repository {
    fun insert(model: T): Boolean

    fun update(model: T): Boolean

    fun get(id: Long): T?

    fun delete(model: T): Boolean

    fun query(): MutableList<T>?
}